### DISCUSSION TITLE
vi/models/fast-sam/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#10048] [Docs]
# vi/models/fast-sam/

Khám phá FastSAM, một giải pháp dựa trên CNN để phân đoạn đối tượng theo thời gian thực trong hình ảnh. Tăng cường tương tác người dùng, hiệu quả tính toán và có thể thích ứng giữa các tác vụ thị giác.

https://docs.ultralytics.com/vi/models/fast-sam/

**Comment by nguyenvandang1998** [comment]
Hi! I am using the FastSAM model and trying to obtain information about the bounding box and mask image of the object, but I don't know how to do it. I hope you can help me.
Thank you very much!!

**Reply by nguyenvandang1998** [reply]
Currently, I can obtain information about the bounding box, but I still don't know how to get the mask image!!

**Reply by glenn-jocher** [reply]
Hi there! 🙌 To extract both bounding boxes and mask images using the FastSAM model, you would typically work with the `results` object returned by the model's prediction method. For the bounding boxes, you can access them directly as shown in the documentation or examples provided. For the mask images, since FastSAM is designed for real-time segmentation, the results should also include segmentation masks corresponding to detected objects.

Here's a simplified way to access both:

```python
from ultralytics import YOLO

# Define your image source
source = 'path/to/your/image.jpg'

# Load the FastSAM model (ensure you've chosen the correct model path)
model = YOLO('FastSAM-s.pt')  # or another appropriate model

# Perform inference
results = model(source)

# Accessing bounding boxes (assuming XYXY format)
boxes = results.boxes.xyxy

# Accessing segmentation masks
# The masks are directly associated with detected objects in the results
masks = results.masks

# To visualize or further process masks and boxes, you would iterate through these objects
```

To visualize or save the masks, you'd typically iterate over the `masks` object, and each mask can be processed according to your requirements (e.g., overlaying on the original image or saving as separate mask images). 

Remember, the exact attribute names (`boxes`, `masks`) and methods might vary based on the model output specification, so it's always good to refer to the specific documentation or example codes for guidance.

Hope this helps! Let us know if you have more questions. 😊
